WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/Reader Messages/PointOfSaleCardPresentPaymentSuccessMessageView.swift

```diff
@@ -13,6 +13,7 @@ struct PointOfSaleCardPresentPaymentSuccessMessageView: View {
     var body: some View {
         if isShowingSendReceiptView {
             POSSendReceiptView(isShowingSendReceiptView: $isShowingSendReceiptView)
+                .transition(.move(edge: .bottom))
```

Comment:
One thing that I learned is to put both animated views within a single view hierarchy so `SwiftUI` would understand better how to animate transitions between them. For example, `VStack`. 

That could potentially be tried out is putting `POSSendReceiptView` and the rest of the success view within a single view hierarchy, for example, `VStack`, or `HStack`. You can see in the `PointOfSaleDashboardView`, how having an `HStack` allows `TotalsView` to slide in together while `CartView` slides out.

Reply:
You can also play around with `geometryGroup` (and `geometryGroupIfSupported` extension I added).

It signals `SwiftUI` to treat some collection of views as one, making them animate together.

Reply:
Wrapping them in a [common stack](6cc198dcef3d19fa48695f57b3dab178ebb108a6) seems to improve it a bit overall. I've also tried both `geometryGroup` and `matchedgeometry` but ends with a very odd results:

https://github.com/user-attachments/assets/62deb6a3-4872-4cfc-9908-60294059c9e0

https://github.com/user-attachments/assets/0b22a3fe-6db5-486e-83cb-2e38a93df59d